clang/include/clang/AST/OperationKinds.def

```diff
@@ -370,6 +370,9 @@ CAST_OPERATION(HLSLArrayRValue)
 // Aggregate by Value cast (HLSL only).
 CAST_OPERATION(HLSLElementwiseCast)
 
+// Splat cast for Aggregates (HLSL only).
+CAST_OPERATION(HLSLSplatCast)
```

Comment:
nit: can we call this `HLSLAggregateSplatCast` I know it is a bit mouthy, but it clarifies the difference between this and the `VectorSplat` cast that already exists.

Reply:
This cast technically also splats vectors in 1 very specific case, but I guess vectors are aggregates :) even if not in the clang code.